eva-server/src/main/java/uk/ac/ebi/eva/server/ws/StudyWSServerV2.java

```diff
@@ -0,0 +1,102 @@
+
+        int totalNumberOfResults = variantStudySummaryService.countAll();
+        if (totalNumberOfResults == 0) {
+            return new ResponseEntity(new PagedResources<>(Collections.EMPTY_LIST, new PagedResources.PageMetadata
+                    (pageSize, pageNumber < 0 ? 0 : pageNumber, totalNumberOfResults)), HttpStatus.NO_CONTENT);
+        }
+
+        List<VariantStudySummary> uniqueStudies = variantStudySummaryService.findAll(pageNumber, pageSize);
+
+        PagedResources.PageMetadata pageMetadata;
+        try {
+            pageMetadata = buildPageMetadata(pageSize, pageNumber, totalNumberOfResults);
+        } catch (IllegalArgumentException e) {
+            return new ResponseEntity(e.getMessage(), HttpStatus.REQUESTED_RANGE_NOT_SATISFIABLE);
+        }
+
+        PagedResources pagedResources = buildPagedResources(uniqueStudies, species, assembly, pageMetadata);
+        return new ResponseEntity(pagedResources, HttpStatus.OK);
+    }
+
+    private PagedResources.PageMetadata buildPageMetadata(Integer pageSize, Integer pageNumber, Integer totalNumberOfResults)
+            throws IllegalArgumentException {
+        Long totalPages = pageSize == 0L ? 0L : (long) Math.ceil((double) totalNumberOfResults / (double) pageSize);
+
+        if (pageNumber < 0 || pageNumber >= totalPages) {
+            throw new IllegalArgumentException("For the given page size, there are " + totalPages + " page(s), so the" +
+                    " correct page range is from 0 to " + String.valueOf(totalPages - 1) + " (both included).");
+        }
+        return new PagedResources.PageMetadata(pageSize, pageNumber, totalNumberOfResults, totalPages);
+    }
+
+    private PagedResources buildPagedResources(List<VariantStudySummary> uniqueStudies, String species,
+                                               String assembly, PagedResources.PageMetadata pageMetadata) {
+
+        PagedResources pagedResources = new PagedResources<>(uniqueStudies, pageMetadata);
+
+        int pageNumber = (int) pageMetadata.getNumber();
+        int pageSize = (int) pageMetadata.getSize();
+
+        if (pageNumber > 0) {
+            pagedResources.add(createPaginationLink(species, assembly, pageNumber - 1, pageSize, "prev"));
+
+            pagedResources.add(createPaginationLink(species, assembly, 0, pageSize, "first"));
+        }
+
+        if (pageNumber < (pageMetadata.getTotalPages() - 1)) {
+            pagedResources.add(createPaginationLink(species, assembly, pageNumber + 1, pageSize, "next"));
+
+            pagedResources.add(createPaginationLink(species, assembly, (int) pageMetadata.getTotalPages() - 1, pageSize, "last"));
+        }
+        return pagedResources;
+    }
+
+    private Link createPaginationLink(String species, String assembly, int pageNumber, int pageSize, String linkName) {
+        return new Link(linkTo(methodOn(ArchiveWSServerV2.class).getBrowsableStudies(species, assembly,
```

Comment:
travis is failing because this class name wasn't updated
```suggestion
        return new Link(linkTo(methodOn(StudyWSServerV2.class).getBrowsableStudies(species, assembly,
```